hortonworks/cloudbreak
# CDPSDX-3525: Added re-registration of cluster proxy config to reattach process for DL resize recovery.

Jira: https://jira.cloudera.com/browse/CDPSDX-3525

**Original Problem**:

- DL resize recovery was timing out due to not being able to communicate with the cluster proxy service
- We determined that this was probably due to the configuration for the DL's stack being deleted after the resized DL is deleted during recovery (the CRN is retained throughout the entire resize process)

**Code Change Objective**:

- Add a step in the reattach logic (used during resize recovery after deletion of the resized DL) which re-registers the cluster proxy configuration for the DL's stack
- This effectively required the creation of a new set of methods to get this to work from the DL code since the cluster proxy service is in the _core_ code

Just to reiterate, the entire point of this change is simply to call the **reRegister** method in the cluster proxy service. We figured the best way to do this in order to follow the pre-existing standard was to invoke a new flow to do this.

Tests done:

- Ensure local version of my code changes for this with CCM V2 does not have the original issue 
- Ensure cluster repair, which runs the upscale flow I modified within my changes, is still successful 
- Ensure reattach after failure in detach (which will now call the new cluster proxy method) is still successful 

Please let me know if you have any questions.
